astrobase/varbase/lcfit.py

```diff
@@ -1456,3 +1467,343 @@ def gaussianeb_fit_magseries(times, mags, errs,
+            It's "fixed", because once you choose your limb-darkening model,
+            it's fixed.
+
+    kwargs:
+        trueparams (list): true parameter values you're fitting for, if they're
+        known (e.g., a known planet, or fake data). Only for plotting on
+        cornerplot.
+
+        burninpercent (float): percent of samples to discard as burn-in.
+
+        useavailablesamples (bool): True if you've already run this, and just
+        want to make the plots. False if you want to run the model fitting.
+
+        plotcorner (bool/str): path to a saved corner plot of the parameters
+        you fit.
+
+        samplesavpath (str): path to save MCMC samples
+
+        magsarefluxes (bool): currently only implemented if True
+
+        n_walkers (int): number of walkers
+
+        n_mcmc_steps (int): number of MCMC steps
+
+        eps (float): for initializing for MCMC step
+
+    returns:
+
+        returndict =  {
+            'fittype':'mandelagoltransit',
+            'fitinfo':{
+                'initialparams':modelparams,
+                'fixedparams':fixedparams,
+                'finalparams':None,
+                'finalparamerrs':None,
+                'leastsqfit':leastsqfit,
+                'fitmags':None,
+                'fitepoch':None,
+            },
+            'fitchisq':npnan,
+            'fitredchisq':npnan,
+            'fitplotfile':None,
+            'magseries':{
+                'phase':None,
+                'times':None,
+                'mags':None,
+                'errs':None,
+                'magsarefluxes':magsarefluxes,
+            },
+        }
+
+    '''
+
+    if not magsarefluxes:
+        raise NotImplementedError
+    if not mandel_agol_dependencies:
+        raise AssertionError(
+            'This function depends on BATMAN, emcee>3.0, and corner.'
+        )
+
+    # sigma clip and get rid of zero errs
+    stimes, smags, serrs = sigclip_magseries(times, mags, errs,
+                                             sigclip=sigclip,
+                                             magsarefluxes=magsarefluxes)
+    nzind = npnonzero(serrs)
+    stimes, smags, serrs = stimes[nzind], smags[nzind], serrs[nzind]
+
+    # parse the input modelparams
+    init_period, init_epoch, init_depth = modelparams[0:3]
+    init_sma, init_incl = modelparams[3:5]
+    fix_ecc, fix_omega = fixedparams[0:2]
+    limb_dark = fixedparams[2]
+    init_u = modelparams[5]
+    init_rp = npsqrt(init_depth)
+
+    # initialize the model and calculate the initial model light-curve
+    init_params, init_m = _transit_model(stimes, init_epoch, init_period,
+                                         init_rp, init_sma, init_incl, fix_ecc,
+                                         fix_omega, init_u, limb_dark)
+    init_flux = init_m.light_curve(init_params)
+
+    # guessed initial params. give nice guesses, or else emcee struggles.
+    theta = [init_rp, init_u[0], init_epoch]
```

Comment:
This is probably also something for later: you can try using scipy.optimize with the transit model to find a decent chi-squared minimum to start emcee and provide these guesses.